test/bad-arguments.test.ts

```diff
@@ -0,0 +1,317 @@
+const URI_DIRECTIVES = ['report-to', 'report-uri'];
+const REQUIRESRIFOR_DIRECTIVES = ['require-sri-for'];
+const SANDBOX_DIRECTIVES = ['sandbox'];
+
+function assertThrowsWithArg (arg: 'NO_ARGUMENTS' | CSPOptions, expectedMessage: string) {
+  expect(() => {
+    if (arg === 'NO_ARGUMENTS') {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (csp as any)();
+    } else {
+      csp(arg);
+    }
+  }).toThrow(expectedMessage);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function assertThrowsWithDirective (directiveKey: string, directiveValue: any, expectedMessage: string) {
+  const arg = {
+    directives: {
+      [directiveKey]: directiveValue,
+    },
+  };
+  assertThrowsWithArg(arg, expectedMessage);
+}
+
+describe('with bad arguments', () => {
+  describe('missing directives', () => {
+    it('errors without arguments', () => {
+      assertThrowsWithArg('NO_ARGUMENTS', 'csp must be called with an object argument. See the documentation.');
+    });
+
+    it('errors with non-objects', () => {
+      [
+        undefined,
+        null,
+        true,
+        false,
+        0,
+        1,
+        '',
+        'str',
+        [],
+        [{ directives: {} }],
+      ].forEach((arg) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        assertThrowsWithArg(arg as any, 'csp must be called with an object argument. See the documentation.');
+      });
+    });
+
+    it('errors with an object that has no "directives" key', () => {
+      assertThrowsWithArg({}, 'csp must have at least one directive under the "directives" key. See the documentation.');
+      assertThrowsWithArg({ reportOnly: true }, 'csp must have at least one directive under the "directives" key. See the documentation.');
+    });
+
+    it('errors with an empty directives object', () => {
+      assertThrowsWithArg({ directives: {} }, 'csp must have at least one directive under the "directives" key. See the documentation.');
+    });
+
+    it('errors with an invalid directive type', () => {
+      assertThrowsWithArg({
+        directives: { 'invalid-directive': ['http://example.com'] },
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } as any, '"invalid-directive" is an invalid directive. See the documentation for the supported list. Force this by enabling loose mode.');
+    });
+  });
+
+  it('tests all directives', () => {
+    const actualDirectives = Object.keys(config.directives);
+    const expectedDirectives = SOURCELIST_DIRECTIVES.concat(
+      BOOLEAN_DIRECTIVES,
+      PLUGINTYPE_DIRECTIVES,
+      URI_DIRECTIVES,
+      REQUIRESRIFOR_DIRECTIVES,
+      SANDBOX_DIRECTIVES
+    );
+    expect(actualDirectives.sort()).toStrictEqual(expectedDirectives.sort());
+  });
+
+  SOURCELIST_DIRECTIVES.forEach((directive) => {
+    [directive, camelize(directive)].forEach((key) => {
+      describe(`${key } directive, a source list`, () => {
```

Comment:
Nitpick: should there be a space after `key` here?

Reply:
Ah, sorry, usually Prettier would catch this for me!